test/e2e/base_user_test.go

```diff
@@ -100,6 +101,7 @@ func newBannedUser(host *wait.HostAwaitility, email string) *v1alpha1.BannedUser
 func (s *baseUserIntegrationTest) deactivateAndCheckUser(userSignup *v1alpha1.UserSignup, mur *v1alpha1.MasterUserRecord) {
 	userSignup, err := s.hostAwait.UpdateUserSignupSpec(userSignup.Name, func(us *v1alpha1.UserSignup) {
 		us.Spec.Deactivated = true
+		states.SetDeactivating(us, true)
```

Comment:
It's "technically" not required, as the UserSignup controller will still go ahead with the deactivation even if the `deactivating` state isn't set.  I'm setting it here for consistency though, as the general deactivation workflow as per the deactivation controller is 1) set deactivating true -> 2) wait for notification -> 3) wait for deactivation time to arrive -> 4) set deactivation true.  Strictly speaking though you're right, we don't need it and I can remove it from here if you think it's confusing.

Reply:
We should be able to deactivate a UserSignup by just setting Spec.Deactivated = true. So, let's remove this line. It will prove that it works as expected.

Reply:
Done